packages/cashier/src/stores/withdraw-store.ts

```diff
@@ -290,32 +295,28 @@ export default class WithdrawStore {
                 decimals: getDecimalPlaces(currency),
             });
             if (!is_ok) error_message = message;
-            else if (+balance < +converter_from_amount) error_message = localize('Insufficient funds');
-            else if (+balance < +min_withdraw_amount) {
+            else if (Number(balance) < Number(converter_from_amount)) error_message = localize('Insufficient funds');
+            else if (min_withdraw_amount && Number(balance) < Number(min_withdraw_amount)) {
                 error_message = localize(
                     'Your balance ({{format_balance}} {{currency}}) is less than the current minimum withdrawal allowed ({{format_min_withdraw_amount}} {{currency}}). Please top up your account to continue with your withdrawal.',
                     { format_balance, currency, format_min_withdraw_amount }
                 );
-            } else if (+converter_from_amount < +min_withdraw_amount || +converter_from_amount > +max_withdraw_amount) {
+            } else if (
+                (min_withdraw_amount && Number(converter_from_amount) < Number(min_withdraw_amount)) ||
+                Number(converter_from_amount) > Number(max_withdraw_amount)
+            ) {
                 error_message = localize(
                     'The current allowed withdraw amount is {{format_min_withdraw_amount}} to {{format_max_withdraw_amount}} {{currency}}',
                     { format_min_withdraw_amount, format_max_withdraw_amount, currency }
                 );
             }
 
-            if (isMobile() && error_message.length > 35) {
+            if (isMobile() && (error_message as string).length > 35) {
                 const error_content = error_message;
-                error_message = (
-                    <ReadMore
-                        expand_text={localize('more')}
-                        text={error_content}
-                        collapse_length={28}
-                        openDialog={() =>
-                            this.error.setErrorMessage({ code: 'CryptoWithdrawalReadMore', message: error_content })
-                        }
-                        show_dialog
-                    />
-                );
+                const openDialog = () => {
```

Comment:
Why should it be a function? can't be something like this:

```suggestion
                const openDialog = this.error.setErrorMessage({ code: 'CryptoWithdrawalReadMore', message: error_content });
```

Reply:
only on click of read more link, the function needs to be called, so we need to have it inside function only